src/localplayer.cpp

```diff
@@ -1057,49 +1060,20 @@ void LocalPlayer::old_move(f32 dtime, Environment *env, f32 pos_max_d,
 
 float LocalPlayer::getSlipFactor(Environment *env, const v3f &speedH)
 {
-
-	if (!touching_ground)
-		return 1.0f;
-
-	float slip_factor = 1.0f;
 	// Slip on slippery nodes
 	const NodeDefManager *nodemgr = env->getGameDef()->ndef();
 	Map *map = &env->getMap();
 	const ContentFeatures &f = nodemgr->get(map->getNodeNoEx(
-			floatToInt(getPosition() - v3f(0, 0.05f * BS, 0), BS)));
+			getStandingNodePos()));
 	int slippery = 0;
-	if (f.walkable) {
+	if (f.walkable)
 		slippery = itemgroup_get(f.groups, "slippery");
-	} else if (is_slipping) {
-		// slipping over an edge? Check surroundings for slippery nodes
```

Comment:
This entire block exists to "correctly" handle the case where you run full-speed off the edge of an ice floe and suddenly come to a stop at the edge:

```
-->
III
GGGGG
```
where `I` is ice and `G` is grass or other unslippery. Is this feature unnecessary, or is this case handled by `getStandingNodePos`? Or am I missing something?

Reply:
We'll test this.

Reply:
@bendeutsch As long the collision detects one or multiple nodes which prevent the player from falling (Yspeed = 0), the closest one will be used for the standing node position.

Reply:
Right, you had said above that `getStandingNodePos` had been changed, so that should handle it nicely.

… brb while I check that this change made it to #6219 – that was branched in August 2017 :-P